File: backend/models/group.py
from sqlalchemy import Column, ForeignKey, Integer, String, DateTime
from db.base_class import Base
from datetime import datetime
from sqlalchemy.orm import relationship


class Group(Base):
    __tablename__ = "group"

    id = Column(Integer, primary_key=True, index=True)  # PK

    name = Column(String, nullable=False)  # 그룹명
    hex_color = Column(String, nullable=False)  # 그룹 색상
    explanation = Column(String, nullable=True)  # 그룹 설명

    create_date = Column(DateTime(timezone=True), nullable=False, default=datetime.now)

    user_id = Column(Integer, ForeignKey("user.id"))
    user = relationship("User", back_populates="group")

    workers = relationship("Worker", back_populates="group")

    def __repr__(self):
        return f"<Group id={self.id}, name={self.name}, hex_color={self.hex_color}, explanation={self.explanation}, create_date={self.create_date}, user_id={self.user_id}>"

File: src/types/group.ts
import { UserData } from './user';

/** 그룹 데이터 인터페이스 */
export interface GroupData {
  id: string;
  name: string;
  hexColor: string;
  explanation?: string;
  createDate?: string;
  user?: UserData;
}

/** 그룹 생성 API 바디 */
export type GroupCreateBody = {
  userId?: string;
} & Pick<GroupData, 'name' | 'hexColor' | 'explanation'>;

/** 그룹 업데이트 API 바디 */
export type GroupUpdateBody = Partial<GroupCreateBody>;
